packages/react-components/src/shared/src/forwardRef.ts

```diff
@@ -1,12 +1,17 @@
 // Idea and naming based off Chakra UI. https://github.com/chakra-ui/chakra-ui/blob/main/packages/system/src/system.types.tsx
 import { ComponentProps, ElementRef, ElementType, ForwardRefExoticComponent, ForwardRefRenderFunction, HTMLProps, RefAttributes, WeakValidationMap, forwardRef as reactForwardRef } from "react";
 
-type UnwrapElementType<T> = T extends ElementType ? ElementRef<T> : T;
+type AsRef<T> = T extends ElementType ? ElementRef<T> : T;
 
-type PropsOf<T> =
-    T extends ElementType ? HTMLProps<UnwrapElementType<T>> & ComponentProps<T> & RefAttributes<UnwrapElementType<T>> :
+type PropsWithoutChildren<P> =
```

Comment:
In some cases, we didn't want to allow the children props, since we don't allow it, but the as type was bringing it back

Reply:
Looks like a good comment to add in the code instead of the review ;)

Reply:
Is it really without? Seems to me more like `PropsWithOptionalChildren`